### PR TITLE
Adding History Shard term

### PR DESCRIPTION
**Pull request by flossypurse**
- breaks up services into terms
- adds History Shard term

### CODE REVIEW

**Review by djmagee — APPROVED** [review]

File: docs/clusters.md [hunk 1]
```diff
@@ -231,9 +253,9 @@ It talks to the Frontend Service, History Service, and the database.
 - It uses grpcPort 7235 to host the service handler.
 - It uses port 6935 for membership related communication.
 
-#### Worker Service
+### Worker Service
 
-The Worker Service runs background processing for the replication queue, system Workflows, and (in versions older than 1.5.0) the Kafka visibility processor.
+The Worker Service runs background processing for the eplication queue, system Workflows, and (in versions older than 1.5.0) the Kafka visibility processor.
```
Comment (aarohib) [inline docs/clusters.md:258]:
```suggestion
The Worker Service runs background processing for the replication queue, system Workflows, and (in versions older than 1.5.0) the Kafka visibility processor.
```

**Review by aarohib — APPROVED** [review]

File: docs/concepts/what-is-a-history-service.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+id: what-is-a-history-service
+title: What is a History Service?
+description: The History Service is responsible for persisting Workflow Execution state and determining what to do next to progress the Workflow Execution through History Shards.
+sidebar_label: History Service
+tags:
+  - term
+---
+
+The History Service is responsible for persisting Workflow Execution state and determining what to do next to progress the Workflow Execution through [History Shards](/concepts/what-is-a-history-shard).
```
Comment (djmagee) [inline docs/concepts/what-is-a-history-service.md:10]: Would "...by using History Shards" be an accurate wording change?

File: docs/concepts/what-is-a-history-service.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+![History Service](/diagrams/temporal-history-service.svg)
+
+The total number of History Services may increase between 1 and the total number of History Shards.
+Additionally an individual History Service can support a large number of History Shards.
+Temporal recommends starting at a ratio of 1 History Service for each 500 History Shards.
```
Comment (djmagee) [inline docs/concepts/what-is-a-history-service.md:16]:
```suggestion
The total number of History Services can be between 1 and the total number of History Shards.
An individual History Service can support a large number of History Shards.
Temporal recommends starting at a ratio of 1 History Service for every 500 History Shards.
```

File: docs/concepts/what-is-a-history-service.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+While the total number of History Shards remains static for the life of the Cluster the number of History Services can change.
```
Comment (djmagee) [inline docs/concepts/what-is-a-history-service.md:18]:
```suggestion
Although the total number of History Shards remains static for the life of the Cluster, the number of History Services can change.
```

File: docs/concepts/what-is-a-history-shard.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+sidebar_label: History Shard
+tags:
+  - term
+---
+
+A History Shard is an important unit within a Temporal Cluster by which the scale of concurrent Workflow Execution throughput can be measured.
+
+Each History Shard maps to a single persistence partition.
+A History Shard assumes that there can only be one concurrent operation within a partition at a time.
+In essence the number of History Shards represents the number of concurrent database operations that can occur for a Cluster.
```
Comment (djmagee) [inline docs/concepts/what-is-a-history-shard.md:14]:
```suggestion
A History Shard assumes that only be one concurrent operation can be within a partition at a time.
In essence, the number of History Shards represents the number of concurrent database operations that can occur for a Cluster.
```

File: docs/concepts/what-is-a-history-shard.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+This means that the number of History Shards in a Temporal Cluster plays a significant role in the performance of your Temporal Application.
+
+The total number of History Shards for the Temporal Cluster must be chosen and [set in the Cluster's configuration](/references/configuration#persistence) prior to integrating a database.
+After the Shard count is configured and the database integrated, the total number of History Shards for the Cluster can not be changed.
```
Comment (djmagee) [inline docs/concepts/what-is-a-history-shard.md:18]:
```suggestion
Before integrating a database, the total number of History Shards for the Temporal Cluster must be chosen and set in the Cluster's configuration (see [persistence](/references/configuration#persistence)).
After the Shard count is configured and the database integrated, the total number of History Shards for the Cluster cannot be changed.
```

File: docs/concepts/what-is-a-history-shard.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+In theory there is no limit to the number of History Shards a Temporal Cluster may operate with, but each History Shard adds compute overhead to the Cluster.
+Temporal Clusters have operated successfully using anywhere from 1-64k History Shards, with each Shard responsible for tens of thousands of Workflow Executions.
+However, the right number of History Shards for any given Cluster depends entirely on the Temporal Application that it is supporting and the type of database.
```
Comment (djmagee) [inline docs/concepts/what-is-a-history-shard.md:22]:
```suggestion
In theory, a Temporal Cluster can operate with an unlimited number of History Shards, but each History Shard adds compute overhead to the Cluster.
Temporal Clusters have operated successfully using anywhere from 1 to 64K History Shards, with each Shard responsible for tens of thousands of Workflow Executions.
However, the correct number of History Shards for any given Cluster depends entirely on the Temporal Application that it is supporting and the type of database.
```

File: docs/concepts/what-is-a-history-shard.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+A History Shard is represented as a hashed integer.
+Each Workflow Execution is automatically assigned to a History Shard.
+The assignment algorithm hashes Workflow Execution metadata such as Workflow Id and Namespace and uses that to match a History Shard.
```
Comment (djmagee) [inline docs/concepts/what-is-a-history-shard.md:26]:
```suggestion
The assignment algorithm hashes Workflow Execution metadata such as Workflow Id and Namespace and uses that value to match a History Shard.
```

File: docs/concepts/what-is-a-history-shard.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+Each History Shard maintains the Workflow Execution Event History, Workflow Execution Mutable State, and a set of the following Internal Task Queues:
```
Comment (djmagee) [inline docs/concepts/what-is-a-history-shard.md:28]:
```suggestion
Each History Shard maintains the Workflow Execution Event History, Workflow Execution mutable state, and the following internal Task Queues:
```

File: docs/concepts/what-is-a-history-shard.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+- Internal Transfer Task Queue: Transfers internal tasks to the Matching Service.
+  Whenever a new Workflow Task needs to be scheduled, the History Service's Transer Task Queue Processor transactionally dispatches it to the Matching Service.
```
Comment (djmagee) [inline docs/concepts/what-is-a-history-shard.md:31]:
```suggestion
  Whenever a new Workflow Task needs to be scheduled, the History Service's Transfer Task Queue Processor transactionally dispatches it to the Matching Service.
```

File: docs/concepts/what-is-a-history-shard.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+- Internal Timer Task Queue: Durably persists Timers.
+- Internal Replicator Task queue: Asynchronously replicates Workflow Executions from active Clusters to other passive Clusters (experimental Multi-Cluster feature).
```
Comment (djmagee) [inline docs/concepts/what-is-a-history-shard.md:33]:
```suggestion
- Internal Replicator Task Queue: Asynchronously replicates Workflow Executions from active Clusters to other passive Clusters.
  (Relies on the experimental Multi-Cluster feature.)
```

File: docs/concepts/what-is-a-history-service.md (hunk 2 above)
Comment (flossypurse) [inline docs/concepts/what-is-a-history-service.md:10]:
```suggestion
The History Service is responsible for persisting Workflow Execution state and determining what to do next to progress the Workflow Execution by using [History Shards](/concepts/what-is-a-history-shard).
```

File: docs/concepts/what-is-a-history-service.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+  - term
+---
+
+The History Service is responsible for persisting Workflow Execution state and determining what to do next to progress the Workflow Execution by using [History Shards](/concepts/what-is-a-history-shard).
+
+![History Service](/diagrams/temporal-history-service.svg)
+
+The total number of History Services can be between 1 and the total number of History Shards.
+An individual History Service can support a large number of History Shards.
+Temporal recommends starting at a ratio of 1 History Service for every 500 History Shards.
```
Comment (yiminc) [inline docs/concepts/what-is-a-history-service.md:16]: this seems like duplicate of what is said in clusters.md

File: docs/concepts/what-is-a-history-shard.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+The assignment algorithm hashes Workflow Execution metadata such as Workflow Id and Namespace and uses that value to match a History Shard.
+
+Each History Shard maintains the Workflow Execution Event History, Workflow Execution mutable state, and the following internal Task Queues:
+
+- Internal Transfer Task Queue: Transfers internal tasks to the Matching Service.
+  Whenever a new Workflow Task needs to be scheduled, the History Service's Transfer Task Queue Processor transactionally dispatches it to the Matching Service.
+- Internal Timer Task Queue: Durably persists Timers.
+- Internal Replicator Task Queue: Asynchronously replicates Workflow Executions from active Clusters to other passive Clusters.
+  (Relies on the experimental Multi-Cluster feature.)
+- Internal Visibility Task Queue: Pushes data to the [Advanced Visibility](/concepts/what-is-advanced-visibility) index.
```
Comment (yiminc) [inline docs/concepts/what-is-a-history-shard.md:35]: same here, seems like duplicate of what is said in clusters.md

**Review by yiminc — APPROVED** [review]

**Pull request merged**